plugins/range/plugin.go

```diff
@@ -31,34 +32,26 @@ type Record struct {
 	expires time.Time
 }
 
-// Records holds a MAC -> IP address and lease time mapping
-var Records map[string]*Record
-
-// DHCPv6Records and DHCPv4Records are mappings between MAC addresses in
-// form of a string, to network configurations.
+// various global variables
 var (
-	// TODO change DHCPv6Records to Record
-	DHCPv6Records map[string]*Record
-	DHCPv4Records map[string]*Record
-	LeaseTime     time.Duration
-	filename      string
-	ipRangeStart  net.IP
-	ipRangeEnd    net.IP
+	// Recordsv4 holds a MAC -> IP address and lease time mapping
+	Recordsv4    map[string]*Record
+	Recordsv6    map[string]*Record
+	LeaseTime    time.Duration
+	filename     string
+	ipRangeStart net.IP
+	ipRangeEnd   net.IP
 )
 
-// LoadDHCPv6Records loads the DHCPv6Records global map with records stored on
-// the specified file. The records have to be one per line, a mac address and an
-// IPv6 address.
-func LoadDHCPv6Records(filename string) (map[string]*Record, error) {
-	// TODO load function for IPv6
-	return nil, errors.New("not implemented for IPv6")
-}
-
-// LoadDHCPv4Records loads the DHCPv4Records global map with records stored on
+// loadRecords loads the DHCPv6/v4 Records global map with records stored on
 // the specified file. The records have to be one per line, a mac address and an
-// IPv4 address.
-func LoadDHCPv4Records(filename string) (map[string]*Record, error) {
-	log.Printf("reading leases from %s", filename)
+// IP address.
+func loadRecords(r io.Reader, v6 bool) (map[string]*Record, error) {
```

Comment:
The 1st parameter (io.Reader) is not being used.

Reply:
fixed

Reply:
another indicator that using global variables is rarely a good idea...